numpy/_core/src/common/half.hpp

```diff
@@ -30,7 +30,7 @@ class Half final {
     #if defined(NPY_HAVE_FP16)
         __m128 mf = _mm_load_ss(&f);
         bits_ = _mm_extract_epi16(_mm_cvtps_ph(mf, _MM_FROUND_TO_NEAREST_INT), 0);
-    #elif defined(NPY_HAVE_VSX3) && defined(NPY_HAVE_VSX_ASM)
+    #elif defined(NPY_HAVE_VSX3) && defined(NPY_HAVE_VSX_ASM) && defined(NPY__CPU_TARGET_VSX3)
```

Comment:
As explained in https://numpy.org/devdocs/reference/simd/how-it-works.html#generating-the-main-configuration-header, these `NPY__CPU_TARGET_` defines shouldn't be used. Why is this change necessary?

Reply:
So on other words, adding `&& defined(NPY__CPU_TARGET_VSX3)` will always disables this branch.